api/v1alpha1/stan.go

```diff
@@ -8,4 +19,18 @@ type STAN struct {
 	ClusterID         string        `json:"clusterId,omitempty" protobuf:"bytes,5,opt,name=clusterId"`
 	Subject           string        `json:"subject" protobuf:"bytes,3,opt,name=subject"`
 	SubjectPrefix     SubjectPrefix `json:"subjectPrefix,omitempty" protobuf:"bytes,6,opt,name=subjectPrefix,casttype=SubjectPrefix"`
+	Auth              *STANAuth     `json:"auth,omitempty" protobuf:"bytes,7,opt,name=auth"`
```

Comment:
*  does not need to be a pointer
* use kubebuilder:default to set the default

Note that even if you set a default, you should cater for empty string

Reply:
I always feel weird when I do not set any value for a filed in a spec, but it gets a value after applying it to k8s.